src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -104,6 +105,53 @@ const TopicFragment = ({
     if (searchText.length > 0) fetchData();
   };
 
+  const SECTIONS = [
+    {
+      title: 'First',
+      content: 'Lorem ipsum...'
+    }
+  ];
+
+  const AccordionView = ({data}) => {
+    const [activeSections, setActiveSections] = React.useState([]);
+
+    const renderSectionTitle = () => {
+      return <View style={styles.content}></View>;
+    };
+
+    const renderHeader = (data, index) => {
+      return (
+        <DiscoveryTitleSeparator
+          key="user-title-separator"
+          text="Your Communities"
+          showArrow
+          rotateArrow={activeSections?.some((actived) => actived === index)}
+        />
+      );
+    };
+
+    const renderContent = () => {
+      return (
+        <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
+      );
+    };
+
+    const updateSections = (activeSectionsParams) => {
+      setActiveSections(activeSectionsParams);
+    };
+
+    return (
+      <Accordion
+        sections={SECTIONS}
+        activeSections={activeSections}
+        renderSectionTitle={renderSectionTitle}
+        renderHeader={renderHeader}
+        renderContent={renderContent}
+        onChange={updateSections}
+      />
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of the `AccordionView` component and the `SECTIONS` constant is a significant change. It seems like you are using a hardcoded section for testing purposes. Remember to replace it with actual data. Also, ensure that the `data` prop passed to `AccordionView` is in the correct format expected by the component.

In the `renderHeader` function, you're checking if the current index exists in the `activeSections` array to decide whether to rotate the arrow or not. This is a good use of the `some` method but remember that it will return as soon as it finds a match which might not be what you want if you have multiple active sections.

In the `renderContent` function, you're mapping over `data` and calling `renderItem`. Make sure `renderItem` is defined and handles the data correctly.

Lastly, in the `updateSections` function, you're updating the state directly with the incoming parameter. Be aware that this could lead to unexpected behavior if the incoming parameter is not in the expected format or type.
```diff
+   const SECTIONS = [
+     {
+       title: 'First',
+       content: 'Lorem ipsum...'
+     }
+   ];
+
+   const AccordionView = ({data}) => {
+     const [activeSections, setActiveSections] = React.useState([]);
+
+     const renderSectionTitle = () => {
+       return <View style={styles.content}></View>;
+     };
+
+     const renderHeader = (data, index) => {
+       return (
+         <DiscoveryTitleSeparator
+           key="user-title-separator"
+           text="Your Communities"
+           showArrow
+           rotateArrow={activeSections?.some((actived) => actived === index)}
+         />
+       );
+     };
+
+     const renderContent = () => {
+       return (
+         <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
+       );
+     };
+
+     const updateSections = (activeSectionsParams) => {
+       setActiveSections(activeSectionsParams);
+     };
+
+     return (
+       <Accordion
+         sections={SECTIONS}
+         activeSections={activeSections}
+         renderSectionTitle={renderSectionTitle}
+         renderHeader={renderHeader}
+         renderContent={renderContent}
+         onChange={updateSections}
+       />
+     );
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->